packages/hardhat-plugin/src/index.ts

```diff
@@ -273,6 +314,31 @@ ignitionScope
             hre.config.networks[hre.network.name]?.ignition.maxFeePerGasLimit,
         });
 
+        try {
+          await hre.network.provider.send("hardhat_setLedgerOutputEnabled", [
+            true,
+          ]);
+
+          hre.network.provider.off("connection_start", () =>
+            executionEventListener.ledgerConnectionStart()
+          );
```

Comment:
The `.off` here won't work as anonymous function is not the same instance as was passed into `.on`.

I suspect that the anonymous functions should be replaced with named local functions, that way both `.on` and `.off` can be passed the same instances.

Reply:
would this work?

```ts
hre.network.provider.off("connection_start", executionEventListener.ledgerConnectionStart);
```

Reply:
Yes